Userland/Services/WebContent/WebContentConsoleClient.cpp

```diff
@@ -26,28 +26,31 @@ WebContentConsoleClient::WebContentConsoleClient(JS::Console& console, JS::Realm
     auto& vm = realm.vm();
     auto& window = static_cast<Web::HTML::Window&>(realm.global_object());
 
-    auto console_global_object = realm.heap().allocate_without_realm<ConsoleGlobalObject>(realm, window);
-
-    // NOTE: We need to push an execution context here for NativeFunction::create() to succeed during global object initialization.
-    auto& eso = Web::Bindings::host_defined_environment_settings_object(realm);
-    vm.push_execution_context(eso.realm_execution_context());
-    console_global_object->initialize(realm);
-    vm.pop_execution_context();
-
-    m_console_global_object = JS::make_handle(console_global_object);
+    m_console_execution_context = MUST(JS::Realm::initialize_host_defined_realm(
+        vm, [&](JS::Realm& realm) {
+            auto console_global_object = realm.heap().allocate_without_realm<ConsoleGlobalObject>(realm, window);
+            console_global_object->initialize(realm);
```

Comment:
InitializeHostDefinedRealm initializes the global object you create here in step 11 (callling this function is step 8

Reply:
Thanks, you're right. Doesn't unfortunately fix anything but good to know anyway. :^)